config/initializers/rack_attack.rb

```diff
@@ -71,4 +71,32 @@ def self.auth_token(request)
   path = payload.fetch(:request).fullpath
 
   Rails.logger.warn("[rack-attack] Throttled request #{request_id} from #{ip} to '#{path}'")
+
+  # Web requests are sent to BigQuery via a concern in the ApplicationController.
+  # If Rack intercepts the request it won't reach the controller, so we need
+  # to manually send web requests that are rate limited to BigQuery.
+  if DfE::Analytics.enabled?
+    request = ActionDispatch::Request.new(payload[:request].env)
+    response = ActionDispatch::Response.new(429)
+
+    # Fetch API user
+    token = request.authorization.to_s.split("Bearer ").last
+    if token.present?
+      current_api_token = ::API::TokenManager.find_lead_provider_api_token(token:)
+      user = current_api_token&.lead_provider
+    end
+
+    # Fetch user from session
+    user ||= Sessions::User.from_session(request.session["user_session"])
+
+    rate_limit_event = DfE::Analytics::Event.new
```

Comment:
I think we wanted another event for persist api as well? we will only now send a web request and no `persist_api_request` for 429s?